File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/response/ResponseEntity.java
package com.roxiemobile.networkingapi.network.rest.response;

import com.roxiemobile.networkingapi.network.http.HttpStatus;
import com.roxiemobile.networkingapi.network.http.MediaType;
import com.roxiemobile.networkingapi.network.rest.request.RequestEntity;

import org.jetbrains.annotations.NotNull;

public interface ResponseEntity<T> extends RequestEntity<T> {

// MARK: - Properties

    /**
     * TODO
     */
    @NotNull HttpStatus getHttpStatus();

    /**
     * TODO
     */
    @NotNull MediaType getMediaType();
}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/response/ResponseEntity.kt
@file:Suppress("RedundantVisibilityModifier")

package com.roxiemobile.networkingapi.network.rest.response

import com.roxiemobile.networkingapi.network.http.HttpStatus
import com.roxiemobile.networkingapi.network.http.MediaType
import com.roxiemobile.networkingapi.network.rest.request.RequestEntity

interface ResponseEntity<TBody>: RequestEntity<TBody> {

// MARK: - Properties

    /**
     * TODO
     */
    val httpStatus: HttpStatus

    /**
     * TODO
     */
    val mediaType: MediaType

// MARK: - Methods

    /**
     * TODO
     */
    public override fun clone(): ResponseEntity<TBody>
}
